akka-actor-typed/src/main/scala/akka/actor/typed/internal/receptionist/LocalReceptionist.scala

```diff
@@ -47,6 +49,12 @@ private[akka] object LocalReceptionist extends ReceptionistBehaviorProvider {
   final case class SubscriberTerminated[T](key: ServiceKey[T], ref: ActorRef[ReceptionistMessages.Listing[T]])
       extends InternalCommand
 
+  val terminationWatchers = new mutable.HashMap[String, ActorRef[Nothing]]();
```

Comment:
Mutable state in the `object` is a not ok.
Must be part of `private def behavior` params.

Prefer immutable `Map`, doesn't look like there is a performance reason for using mutable Map.